src/EditorFeatures/TestUtilities/Workspaces/TestWorkspace.cs

```diff
@@ -666,6 +666,13 @@ public override void OpenDocument(DocumentId documentId, bool activate = true)
             testDocument.GetOpenTextContainer();
         }
 
+        /// <summary>
+        /// Overriding base impl so that when we close a document it goes back to the initial state when the test
+        /// workspace was loaded, throwing away any changes made to the open version.
+        /// </summary>
+        internal override void TryOnDocumentClosed(DocumentId documentId)
+            => CloseDocument(documentId);
```

Comment:
this is the impl for a test workspace.  in a mutating workspace (if we ever get such a thing) we could potentially go back to reading from disk.

For a non-mutating workspace, we never get here as the host will handle file closes.